src/locktower.rs

```diff
@@ -0,0 +1,172 @@
+use std::collections::VecDeque;
+pub struct VoteLock {
+    /// Fork id, same as slot id. There can only be 1 fork per slot, since the empty PoH ticks
```

Comment:
A validator should be voting every block, not once per slot. Can there be multiple commitments to the same slot ID (one for each block)?

Reply:
@garious @rob-solana i think as we discussed a slot can only have 1 valid data block that is votable.  A leader could be scheduled for multiple consecutive slots, and that is consensus and vote agnostic.

Reply:
No, that doesn't work for confirmation.  When there's no forks, a validator should be voting on every block.  When there are forks, it should withhold its vote if:

* the forks are tied in weight
* it needs to change forks, but is locked out

Reply:
@garious, what Toly describes is fine, but when we use "slot" to mean something votable, we need to make up a word for "the number of slots for which a leader is scheduled to transmit consecutively".

@carllin what we discussed this morning is potentially borken under this terminology...

Reply:
A block is the thing that's votable. The term is used for nothing else.

Reply:
@garious i think the problem is that it's also a period of time/range of ledger height.  A leader proposes a PoH ledger that spans multiple blocks, but only the last block has data, and that last block is at the height that the leader is scheduled for.

Reply:
https://solana-labs.github.io/book-edge/terminology.html#block
https://solana-labs.github.io/book-edge/terminology.html#slot

So block_id is the `Entry::id` of the last *tick* entry of a block.